src/apps/service/apps/de/listings.clj

```diff
@@ -589,55 +580,72 @@
 
 (defn- format-app-details
   "Formats information for the get-app-details service."
-  [username details tools admin?]
-  (let [app-id                       (:id details)
-        {version-id :id
-         :keys      [app_references]
-         :as        version-details} (-> details :app_versions first)]
+  [{username :shortUsername :as user} details tools app-references admin?]
+  (let [{app-id :id version-id :version_id} details]
     (-> details
-        (select-keys [:id :wiki_url])
-        (merge (select-keys
-                 version-details
-                 [:version
-                  :deleted
-                  :disabled
-                  :edited_date
-                  :integration_date
-                  :integrator_name
-                  :integrator_email]))
+        (select-keys [:id
+                      :wiki_url
+                      :version
+                      :version_id
+                      :deleted
+                      :disabled
+                      :edited_date
+                      :integration_date
+                      :integrator_name
+                      :integrator_email
+                      :step_count
+                      :tool_count
+                      :external_app_count
+                      :task_count
+                      :overall_job_type
+                      :average_rating
+                      :total_ratings
+                      :is_favorite])
         (assoc :name                 (:name details "")
                :description          (:description details "")
-               :version_id           version-id
-               :references           (map :reference_text app_references)
+               :versions             (amp/list-app-versions app-id)
+               :references           (map :reference_text app-references)
                :tools                (map format-app-tool tools)
                :job_stats            (format-app-details-job-stats (str app-id) nil admin?)
                :extra                (format-app-extra-info version-id admin?)
-               :documentation        (format-app-documentation app-id username admin?)
+               :documentation        (format-app-documentation app-id version-id user admin?)
                :categories           (get-groups-for-app app-id)
                :suggested_categories (get-suggested-groups-for-app app-id)
                :system_id            c/system-id)
         (format-app-hierarchies username)
         format-wiki-url)))
 
-;; This function was split from `get-app-details` to provide a way for administrative endopints to skip permission
+;; This function was split from `get-app-details` to provide a way for administrative endpoints to skip permission
```

Comment:
endopints and exopints, new measures of beer consumption, possibly somewhat gross?

(I try to find ways of interpreting typos for amusement, sorry, moving along now...)